R/assignment_linter.R

```diff
@@ -67,7 +67,7 @@ assignment_linter <- function(allow_cascading_assign = TRUE, allow_right_assign
       "//EQ_SUB",
       "//EQ_FORMALS"
     ),
-    "[@line1 < following-sibling::*[1]/@line1 or //COMMENT]"
+    "[@line1 < following-sibling::*[1]/@line1 or @line1 = following-sibling::*[1][self::COMMENT]/@line1]"
```

Comment:
would this work as one condition instead?

```
@line1 < following-sibling::*[not(self::COMMENT)][1]/@line1
```

Reply:
or perhaps even

```
@line1 < following-sibling::expr[1]/@line1
```

(less confident)

Reply:
They both pass all of the tests, and I've added one more to test that it doesn't detect trailing assignment in strings. They're both better than the original query. I've updated with the second suggestion; as my xpath knowledge has come from looking at the lintr queries, I'm happy to go with the first one if you're more comfortable with it

Reply:
as long as it passes tests it should be ok